### ISSUE TITLE
New rule: RedundantExplicitOperatorCall

### ISSUE THREAD

**Issue by osipxd** [detekt/detekt#7034]
*This issue is continue of discussion started in https://github.com/detekt/detekt/issues/7019#issuecomment-1979567525*

## Expected Behavior of the rule

The rule should report explicit call of `operator fun` where it is possible to use its operator form. 

**Config:**

```yml
RedundantExplicitOperatorCall:
    enabled: true
    # This is just an example, list of default operators is not defined yet
    operators:
      - get
      - set
      - not
```

**Examples:**

```kotlin
val featureFlags: Map<String, Boolean> = ...

// Noncompliant
val isMyFeatureEnabled = featureFlags.get("myFeature")
if (isMyFeatureEnabled.not()) { ... }

// Compliant
val isMyFeatureEnabled = featureFlags["myFeature"]
if (!isMyFeatureEnabled) { ... }
```

### Exceptions to the rule

- **With nullable types.** Explicit call with nullable type should not be reported, as operators coldn't be called on nullable types (unles it is extension operator for nullable types):
  ```kotlin
  val featureFlags: Map<String, Boolean>? = ...

  featureFlags?.get("myFeature") // OK
  
  // but if we have an extension-function for nullable type
  operator fun Map<String, Boolean>?.get(key: String) = this?.get(key) ?: false

  featureFlags?.get("myFeature") // Not OK
  ```
- **Call on implicit receiver.** For some of operators the only possible way to use it with implicit receiver is to use it with explicit `this`, so maybe such cases shouldn't be reported:
  ```kotlin
  with(featureFlags) {
      if (get("myFeature1")) println("OK")
      if (this["myFeature2"]) println("Also OK")
  }
  ```
- **Call chain.** It may be not appreciated to use operator in a code written in functional style.
  ```kotlin
  sequenceOf(1, 2, 3)
      .filter { it % 2 == 0 }
      .plus(42) // It is an operator but it is OK to use it in call chain
      .forEach(::println)

  // With operator call it could be
  (sequenceOf(1, 2, 3)
      .filter { it % 2 == 0 }
      + 42)
      .forEach(::println)
  ```

## Context

- It might be useful to restrict explicit usage of some operators according to project codestyle (see #7019 for example)
- There already exists the rule promoting `[]` over `get()` and `set()` - [ExplicitCollectionElementAccessMethod](https://detekt.dev/docs/next/rules/style#explicitcollectionelementaccessmethod). This rule could be implemented via the new rule
